x/concentrated-liquidity/swaps.go

```diff
@@ -170,6 +166,12 @@ func (k Keeper) SwapOutAmtGivenIn(
 		return sdk.Coin{}, sdk.Coin{}, sdk.Int{}, sdk.Dec{}, sdk.Dec{}, err
 	}
 
+	// Settles balances between the tx sender and the pool to match the swap that was executed earlier.
+	// Also emits swap event and updates related liquidity metrics
+	if err := k.updatePoolForSwap(ctx, poolI, sender, tokenIn, tokenOut, newCurrentTick, newLiquidity, newSqrtPrice); err != nil {
```

Comment:
We shouldn't need the call to `k.applySwap` above. It's already called within `updatePoolForSwap`. Results in charging gas twice

Reply:
whoops, good catch

Reply:
https://github.com/osmosis-labs/osmosis/pull/4474/commits/55f9785210d1a3c497920a2e25659fc5d927c19c